src/components/button/Button.tsx

```diff
@@ -1,43 +1,45 @@
-// @flow
 import * as React from 'react';
 import classNames from 'classnames';
-import omit from 'lodash/omit';
-
 import LoadingIndicator from '../loading-indicator';
 import RadarAnimation from '../radar';
 
-type Props = {
+export enum ButtonType {
+    BUTTON = 'button',
+    RESET = 'reset',
+    SUBMIT = 'submit',
+}
+export interface ButtonProps {
```

Comment:
Based on the interwebs, not much realistic difference between using `type` or `interface` for the purpose of react props. But since the latter is more extensible, might as well use it

Reply:
neither here nor there. I prefer to use interface only when planning to work with classes as a signal that I'm planning on this participating in OO

Reply:
It looks like `interface` is used for structural sub-typing which is what you want with props usually

https://www.typescriptlang.org/docs/handbook/interfaces.html